Add changelog for Odoc 2.3.0 (#1550)
ocaml/ocaml.org: data/changelog/odoc/2023-10-02-odoc-2.3.0.md

This motivated the odoc and OCaml.org teams to jointly work on a centralised package documentation, that [went live in April 2022](https://discuss.ocaml.org/t/v3-ocaml-org-we-are-live/9747), as part of the new version of OCaml.org.

With documentation for OCaml libraries [readily available on OCaml.org](https://ocaml.org/packages), we now turn our focus on making sure that library authors have the tooling they need to create high-quality documentation.

Our [roadmap](https://github.com/ocaml/odoc/blob/master/ROADMAP.md) highlights some features we believe will make the generated documentation significantly better for readers, and documentation-writing much more pleasant and rewarding.

This release is a significant milestone in implementing the features on our roadmap and is the precursor to a series of upcoming releases. Odoc 2.4.0 will follow shortly and will bring support for search. Stay tuned and follow our progress through the [OCaml Platform newsletter](https://discuss.ocaml.org/tag/platform-newsletter)!

## 🤝 Join Our Mission

> While we are dedicated to developing the best tooling to generate and serve documentation on OCaml.org, creating a well-documented library ecosystem can only be a collective effort. Package authors: we're working hard to give you great tools, but we'll need all your help to create an ecosystem of well-documented libraries for OCaml!

Reviewer: No need to make this about us.
```suggestion
While we are dedicated to developing the best tooling to generate and serve documentation on OCaml.org, creating a well-documented library ecosystem can only be a collective effort.
Package authors: Your help is needed to create an ecosystem of well-documented libraries for OCaml!
```

My English is bad so I'm probably wrong.

Author: The goal was to emphasis the collaborative aspect of the effort. @christinerose what do you think?

If you find that writing documentation for your library isn't as straightforward as you would like, please do share your feedback with us.

## 🌟 Spotlight Features of Odoc 2.3.0

Here are a couple of the new features introduced in Odoc 2.3.0 that we'd like to highlight.

1. **Table Support**

    Table support is the last addition to the odoc language, after

>     [math expressions](https://github.com/ocaml/odoc/pull/886). Odoc supports two markup to include tables: a light one, and a

Reviewer: Tables are not related to math expressions at all and it's weird that it's the first link. I suggest dropping this sentence.

Author: I removed it, thanks!

>     Table support is the last addition to the odoc language, after
>     [math expressions](https://github.com/ocaml/odoc/pull/886). Odoc supports two markup to include tables: a light one, and a

Author: ```suggestion
    Table support is the last addition to the odoc language, and comes with two syntax flavours: a light one, and a
```
